### PR TITLE
Move Settings Window into Modal

### PR DESCRIPTION
**Pull request by devinbinnie**
#### Summary
This PR moves the Settings Window into a modal instead of a separate window. This is a step towards a more unified Desktop App without extraneous windows. A further change will likely re-skin the interface altogether.

Will need a quick QA smoke-test pass on this, as this might mess with the E2E tests (quick check looks like they're still running). Also asking for a quick UX to make sure I've done makes anything significantly worse.

#### Screenshots
image

```release-note
Move Settings Window into Modal
```

### CODE REVIEW

File: src/renderer/components/SettingsPage.tsx [hunk 1]
```diff
@@ -1228,35 +1230,36 @@ class SettingsPage extends React.PureComponent<Props, State> {
         }
 
         return (
-            <div
-                className='container-fluid'
-                style={{
-                    height: '100%',
-                }}
+            <Modal
```
Comment (M-ZubairAhmed) [inline src/renderer/components/SettingsPage.tsx:1233]: dont we use Modals from mattermost/components here as well?

Comment (devinbinnie) [reply src/renderer/components/SettingsPage.tsx:1233]: Not at this time no, but I would absolutely like to. I was planning on looking into those when we redesign the modal to fit @abhijit-singh's new designs.

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by yasserfaraazkhan — APPROVED** [review]
Hi @devinbinnie 
tested this locally.

For e2e, it wont run on forked repo.
Locally, I've checked. We need update `keyboards_shortcuts.js`
- Replace [Line](https://github.com/devinbinnie/desktop/blob/WIP_settings_modal/e2e/specs/settings/keyboard_shortcuts.test.js#L50) with this code,
 `await settingsWindow.click('#react-select-2-option-0');`
- Delete test: https://github.com/devinbinnie/desktop/blob/WIP_settings_modal/e2e/specs/settings/keyboard_shortcuts.test.js#L67

- Delete test: https://github.com/devinbinnie/desktop/blob/WIP_settings_modal/e2e/specs/settings/keyboard_shortcuts.test.js#L84

**Review by abhijit-singh — APPROVED** [review]
Thanks @devinbinnie! Looks good visually and seems to be working fine, but, I noticed a some lag in the opening of the modal and modifying things like the spell check language.  The existing version in a separate window seems faster to open and interact with. 
Not sure if this is known and if it can be improved though.

**Comment by devinbinnie** [comment]
Talked with @yasserfaraazkhan and these tests were passing fine on my machine, so I think we're okay to merge as-is but we'll keep an eye on them during release testing.

@abhijit-singh I didn't see any lag other than what it takes to open a modal, which is the same lag we have opening other modals I think as well. There might be something we can do about that, but I think I'll have to tackle that separately. Maybe something we can do when we update the designs for all the modals :)

**Comment by streamer45** [comment]
@devinbinnie I am seeing a bunch of warnings when opening the modal, just checking with you whether they are expected:

```
07:45:46.202 › [WebContentsEventM...] [renderer] Warning: Encountered two children with the same key, `%s`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.%s null 
    at div
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-bootstrap_esm_Col_js-node_modules_react-bootstrap_esm_FormControl_-404d0b_bundle.js:132:23
    at div
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-bootstrap_esm_Container_js-node_modules_react-bootstrap_esm_Row_js-978772_bundle.js:137:23
    at div
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-bootstrap_esm_Container_js-node_modules_react-bootstrap_esm_Row_js-978772_bundle.js:89:23
    at div
    at div
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:2098:27
    at div
    at div
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:1736:23
    at div
    at Transition (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:3318:30)
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:1317:24
    at DialogTransition
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:2274:24
    at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:1431:23
    at SettingsPage (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/settings_bundle.js:146:5)
    at injectIntl(SettingsPage)
    at IntlProvider (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-intl_lib_src_components_message_js-node_modules_react-intl_lib_src-0be939_bundle.js:5837:47)
    at IntlProvider (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/src_common_utils_constants_ts-src_renderer_intl_provider_tsx_bundle.js:357:5) (vendors-node_modules_react-dom_index_js_bundle.js:176)
```

**Comment by devinbinnie** [comment]
> @devinbinnie I am seeing a bunch of warnings when opening the modal, just checking with you whether they are expected:
> 
> ```
> 07:45:46.202 › [WebContentsEventM...] [renderer] Warning: Encountered two children with the same key, `%s`. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.%s null 
>     at div
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-bootstrap_esm_Col_js-node_modules_react-bootstrap_esm_FormControl_-404d0b_bundle.js:132:23
>     at div
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-bootstrap_esm_Container_js-node_modules_react-bootstrap_esm_Row_js-978772_bundle.js:137:23
>     at div
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-bootstrap_esm_Container_js-node_modules_react-bootstrap_esm_Row_js-978772_bundle.js:89:23
>     at div
>     at div
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:2098:27
>     at div
>     at div
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:1736:23
>     at div
>     at Transition (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:3318:30)
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:1317:24
>     at DialogTransition
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:2274:24
>     at file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_css-loader_dist_cjs_js_node_modules_bootstrap-dark_src_bootstrap-dark_cs-743cec_bundle.js:1431:23
>     at SettingsPage (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/settings_bundle.js:146:5)
>     at injectIntl(SettingsPage)
>     at IntlProvider (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/vendors-node_modules_react-intl_lib_src_components_message_js-node_modules_react-intl_lib_src-0be939_bundle.js:5837:47)
>     at IntlProvider (file:///home/streamer45/go/src/github.com/mattermost/desktop/dist/renderer/src_common_utils_constants_ts-src_renderer_intl_provider_tsx_bundle.js:357:5) (vendors-node_modules_react-dom_index_js_bundle.js:176)
> ```

Yes I believe these are leftovers from the original screen, which I didn't modify much. I do plan on re-doing it once given the time.

**Review by streamer45 — APPROVED** [review]
Looks good!

**Pull request merged**